packages/material-ui/src/Typography/Typography.js

```diff
@@ -1,106 +1,50 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import { useThemeVariants } from '@material-ui/styles';
-import withStyles from '../styles/withStyles';
+import experimentalStyled from '../styles/experimentalStyled';
 import capitalize from '../utils/capitalize';
+import typographyClasses, { getTypographyUtilityClass } from './typographyClasses';
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    margin: 0,
-  },
-  /* Styles applied to the root element if `variant="body2"`. */
-  body2: theme.typography.body2,
-  /* Styles applied to the root element if `variant="body1"`. */
-  body1: theme.typography.body1,
-  /* Styles applied to the root element if `variant="caption"`. */
-  caption: theme.typography.caption,
-  /* Styles applied to the root element if `variant="button"`. */
-  button: theme.typography.button,
-  /* Styles applied to the root element if `variant="h1"`. */
-  h1: theme.typography.h1,
-  /* Styles applied to the root element if `variant="h2"`. */
-  h2: theme.typography.h2,
-  /* Styles applied to the root element if `variant="h3"`. */
-  h3: theme.typography.h3,
-  /* Styles applied to the root element if `variant="h4"`. */
-  h4: theme.typography.h4,
-  /* Styles applied to the root element if `variant="h5"`. */
-  h5: theme.typography.h5,
-  /* Styles applied to the root element if `variant="h6"`. */
-  h6: theme.typography.h6,
-  /* Styles applied to the root element if `variant="subtitle1"`. */
-  subtitle1: theme.typography.subtitle1,
-  /* Styles applied to the root element if `variant="subtitle2"`. */
-  subtitle2: theme.typography.subtitle2,
-  /* Styles applied to the root element if `variant="overline"`. */
-  overline: theme.typography.overline,
-  /* Styles applied to the root element if `variant="inherit"`. */
-  inherit: {},
-  /* Styles applied to the root element if `align="left"`. */
-  alignLeft: {
-    textAlign: 'left',
-  },
-  /* Styles applied to the root element if `align="center"`. */
-  alignCenter: {
-    textAlign: 'center',
-  },
-  /* Styles applied to the root element if `align="right"`. */
-  alignRight: {
-    textAlign: 'right',
-  },
-  /* Styles applied to the root element if `align="justify"`. */
-  alignJustify: {
-    textAlign: 'justify',
-  },
-  /* Styles applied to the root element if `nowrap={true}`. */
-  noWrap: {
+const getTextColor = (color, palette) => {
+  if (color === 'textPrimary' || color === 'textSecondary') {
```

Comment:
Maybe it's better to test if the color starts with `text`? Will help us if we decided to dynamically support more colors

Reply:
Done